be/src/exprs/array_functions.tpp

```diff
@@ -1060,296 +1084,273 @@ private:
+                if (has_data) {
+                    result_ptr->append(result);
                 } else {
-                    LOG(ERROR) << "unhandled types other than arithmetic/time/decimal for sum and avg";
-                    DCHECK(false) << "other types than arithmetic/time/decimal is not support sum "
-                                     "and avg";
-                    result_column->append_default();
+                    result_ptr->append_default();
+                    null_ptr[i] = 1;
                 }
             } else {
-                result_column->append_default();
-                (*null_ptr)[i] = 1;
+                result_ptr->append(result);
             }
         }
-
-        return result_column;
     }
+};
 
-    template <bool IsMin, ArithmeticType type, LogicalType ValueType, bool HasNull, typename ElementColumn>
-    static StatusOr<ColumnPtr> _min_and_max(const ElementColumn& elements, const UInt32Column& offsets,
-                                            const NullColumn::Container* null_elements,
-                                            std::vector<uint8_t>* null_ptr) {
-        const size_t num_array = offsets.size() - 1;
-        auto offsets_ptr = offsets.get_data().data();
+template <bool isSum>
+class ArraySumAvg {
+public:
+    template <LogicalType ResultType, LogicalType ElementType, bool HasNull>
+    static void process(const Column* elements, const NullColumn* elements_null_col, const UInt32Column* offsets,
+                        Column* result_col, NullColumn* null_cols) {
+        const RunTimeCppType<TYPE_NULL>* elements_nulls = nullptr;
+        if constexpr (HasNull) {
+            elements_nulls = elements_null_col->get_data().data();
+        }
 
-        using ResultColumnType = RunTimeColumnType<ValueType>;
+        auto* elements_data = down_cast<const RunTimeColumnType<ElementType>*>(elements)->get_data().data();
+        auto* offsets_ptr = offsets->get_data().data();
+        auto* null_ptr = null_cols->get_data().data();
+        const int64_t num_array = offsets->size() - 1;
 
-        auto result_column = ResultColumnType::create();
-        result_column->reserve(num_array);
+        result_col->reserve(num_array);
+        auto result_ptr = down_cast<RunTimeColumnType<ResultType>*>(result_col);
 
-        auto elements_ptr = (const RunTimeCppType<ValueType>*)(elements.raw_data());
+        using ResultCppType = RunTimeCppType<ResultType>;
+        int scale = 0;
+        if constexpr (lt_is_decimal<ElementType>) {
+            scale = down_cast<const RunTimeColumnType<ElementType>*>(elements)->scale();
+        }
 
         for (size_t i = 0; i < num_array; i++) {
             size_t offset = offsets_ptr[i];
-            size_t array_size = offsets_ptr[i + 1] - offsets_ptr[i];
-
-            if (array_size > 0) {
-                RunTimeCppType<ValueType> result;
+            int64_t array_size = offsets_ptr[i + 1] - offsets_ptr[i];
```

Comment:
a null array's offset size may not 0, please process null explicitly

Reply:
same as above